src/data.js

```diff
@@ -0,0 +1,114 @@
+import * as Render from './renders';
+
+function parseWeatherData(data) {
+  return {
+    city: data.name,
+    temp: data.main.temp,
+    weather: data.weather[0].main,
+    weather_desc: data.weather[0].description,
+    min_temp: data.main.temp_min,
+    max_temp: data.main.temp_max,
+    humidity: data.main.humidity,
+    lat: data.coord.lat,
+    lon: data.coord.lon,
+  };
+}
+
+function convertTemps(data) {
+  const units = document.getElementById('temp-units');
+  switch (units.value) {
+    case 'C':
+      data.temp = Math.round(data.temp - 273.15);
+      data.max_temp = Math.round(data.max_temp - 273.15);
+      data.min_temp = Math.round(data.min_temp - 273.15);
+      break;
+    case 'F':
+      data.temp = Math.round((data.temp * (9 / 5)) - 459.67);
+      data.max_temp = Math.round((data.max_temp * (9 / 5)) - 459.67);
+      data.min_temp = Math.round((data.min_temp * (9 / 5)) - 459.67);
+      break;
+    default:
+      break;
+  }
+
+  return data;
+}
+
+function noCityInfo() {
+  return {
+    city: 'No city found',
+    temp: 0,
+    weather: '404 error',
+    weather_desc: 'No city found',
+    min_temp: 0,
+    max_temp: 0,
+    humidity: 0,
+    lat: 0,
+    lon: 0,
+  };
+}
+
+export async function getWeatherInfo(city) {
+  const key = 'acb20f9ff5d6a288ecb4800af4426da3';
+  let weatherApiURL = 'https://api.openweathermap.org/data/2.5/weather';
+
+  weatherApiURL += `?q=${city}`;
+  weatherApiURL += `&appid=${key}`;
+
+  const response = await fetch(weatherApiURL);
+  const rawData = await response.json();
+
+  return parseWeatherData(rawData);
+}
+
+export async function getImage(type) {
+  let unsplashApiURL = 'https://api.unsplash.com/photos/random';
+  const clientID = 'yzaA27Bnm-RKqmb9ud2qC1imujfvoCREjcqG23C4ca0';
+  const orientation = 'landscape';
+  const count = 1;
+  const fetchOptions = {
+    method: 'GET',
+    headers: {
+      'Accept-Version': 'v1',
+    },
+  };
+
+  unsplashApiURL += `?orientation=${orientation}`;
+  unsplashApiURL += `&count=${count}`;
+  unsplashApiURL += `&client_id=${clientID}`;
+  unsplashApiURL += `&query=${type}`;
+
+  const response = await fetch(unsplashApiURL, fetchOptions);
+  const [realData] = await response.json();
+
```

Comment:
[REQUIRED] This function needs a try/catch block to handle fetching errors (this is not the only async function without error handling )

![image](https://user-images.githubusercontent.com/73254040/96968601-d55b0100-14d6-11eb-8223-118a17d95d5c.png)
[Source](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/)